modules/extension/mbstyle/src/main/java/org/geotools/mbstyle/parse/MBObjectParser.java

```diff
@@ -1426,9 +1440,74 @@ public Displacement displacement(JSONObject json, String tag, Displacement fallb
         }
     }
 
-    /** @return True if the layer has the provided property explicitly provided, False otherwise. */
+    //
+    // structure checks
+    //
+    /** @return True if json has the provided property explicitly provided, False otherwise. */
     public boolean isPropertyDefined(JSONObject json, String propertyName)
             throws MBFormatException {
+        return isDefined(json, propertyName);
+    }
+    /** @return True if json has the property explicitly provided, False otherwise. */
+    public boolean isDefined(JSONObject json, String propertyName) throws MBFormatException {
         return json.containsKey(propertyName) && json.get(propertyName) != null;
     }
+    /** @return True if array has the property explicitly provided, False otherwise. */
+    public boolean isDefined(JSONArray json, int index) throws MBFormatException {
```

Comment:
All the `int index` versions of the `is___` methods would benefit from a modified javadoc - arrays don't have properties, they have indices. Something like
```
/** @return True if array has a value at the provided index, False otherwise. */
```
Would make more sense

Reply:
Thanks, I have rewritten these.